facturacion_mexico/facturacion_fiscal/tests/test_cancel_guard_draft_ffm.py

```
@@ -0,0 +1,217 @@
+	def test_draft_procesando_sigue_bloqueando(self):
+		# Draft en PROCESANDO (sin UUID ni facturapi_id): puede haber operación en curso → BLOQUEA
+		_seed_ffm(self.si, "PROCESANDO", 0)
+		with mock.patch("frappe.model.document.Document.cancel"):
+			with self.assertRaises(frappe.ValidationError):
+				cancelar_si_post_fiscal(self.si)
+
+	def test_draft_borrador_que_es_la_ffm_activa_bloquea(self):
+		# Si la SI apunta a un borrador sin timbrar como FFM activa, nunca se ignora → BLOQUEA
+		frappe.db.set_value("Sales Invoice", self.si, "fm_factura_fiscal_mx", self.ffm_draft)
+		with mock.patch("frappe.model.document.Document.cancel"):
+			with self.assertRaises(frappe.ValidationError):
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)